tests/lib/external/snapd-testing-tools/utils/log-analyzer

```diff
@@ -1,330 +1,220 @@
-    all_tasks="$(list_all_tasks "$exec_exp")"
-    executed_tasks="$(list_executed_tasks "$exec_exp" "$log")"
-    failed_tasks="$(list_failed_tasks "$exec_exp" "$log")"
-    failed_tasks_prepare="$(_list_failed task prepare "$log")"
-    failed_tasks_restore="$(_list_failed task restore "$log")"
-
-    # In case no tasks for the expression, the aborted list is empty
-    if [ -z "$all_tasks" ]; then
-        return
-    fi
-
-    # In case no tasks are successfully executed, all the tasks - the failed ones are the aborted
-    if [ -z "$executed_tasks" ]; then
-        exec_and_failed_tasks="$(_list_executed_and_failed_tasks "$exec_exp" "$log")"
-        _diff_tasks_lists "$all_tasks" "$exec_and_failed_tasks"
-        return
-    fi
-    
-    # In other cases the aborted tasks are all the tasks - the executed - the that failed 
-    _diff_tasks_lists "$all_tasks" "$executed_tasks"
-}
-
-main() {
-    if [ $# -eq 0 ]; then
-        show_help
-        exit 0
-    fi
-
-    local subcommand="$1"
-    local action=
-    while [ $# -gt 0 ]; do
-        case "$1" in
-            -h|--help)
-                show_help
-                exit 0
-                ;;
-            *)
-                action=$(echo "$subcommand" | tr '-' '_')
-                shift
-                break
-                ;;
-        esac
-    done
-
-    if [ -z "$(declare -f "$action")" ]; then
-        echo "log.analyzer: no such command: $subcommand" >&2
-        show_help
-        exit 1
-    fi
-
-    "$action" "$@"
-}
-
-main "$@"
+#!/usr/bin/env python3
+
+# The following script requires Python 3.9 or higher
+
+import argparse
+import json
+import subprocess
+from typing import Callable, TypedDict, Literal, Union
+
+
+class SpreadLog_TypePhase(TypedDict):
+    type: Literal["phase"]
+    task: str
+    verb: str
+
+
+class SpreadLogDetail(TypedDict):
+    lines: list[str]
+
+
+class SpreadLog_TypeResult(TypedDict):
+    type: Literal["result"]
+    result_type: str
+    level: str
+    stage: str
+    detail: SpreadLogDetail
+
+
+SpreadLog = Union[SpreadLog_TypePhase, SpreadLog_TypeResult]
+
+
+def filter_with_spread(exec_param: str) -> list[str]:
+    exec_param = exec_param.replace(",", " ")
+    cmd = ["spread", "-list"]
+    cmd.extend(exec_param.split())
+    return subprocess.check_output(cmd, universal_newlines=True).splitlines()
+
+
+def list_executed_tasks(
+    filtered_exec_param: set[str], spread_logs: list[SpreadLog]
+) -> set[str]:
+    executed = [
+        log["task"]
+        for log in spread_logs
+        if log["type"] == "phase" and log["verb"] == "Executing"
+    ]
+    return filtered_exec_param.intersection(executed)
+
+
+def _get_detail_lines(
+    spread_logs: list[SpreadLog], log_condition_func: Callable[[SpreadLog], bool]
+) -> list[str]:
+    result = [log["detail"]["lines"] for log in spread_logs if log_condition_func(log)]
+
+    def clean_entry(entry: str) -> str:
+        entry = entry.strip()
+        if entry.startswith("-"):
```

Comment:
Perhaps `entry.removeprefix` so that you can chain the whole thing to:
```python
def clean_line(line: str) -> str:
    return line.strip().removeprefix('-').strip()
```

Can you clarify why we strip the leading dash though? Ideally as a code comment.